pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1185,8 +1185,9 @@ func (cb *ClusterBuilder) setUpstreamProtocol(mc *MutableCluster, port *model.Po
 	// Since protocol detection is disabled for gateway and use_downstream_protocol is used
 	// under protocol detection for cluster to select upstream connection protocol when
 	// the service port is unnamed. use_downstream_protocol should be disabled for gateway.
-	if cb.sidecarProxy() && ((util.IsProtocolSniffingEnabledForInboundPort(port) && direction == model.TrafficDirectionInbound) ||
```

Comment:
It seems the current behavior is not work for pure http1.1 server either

Reply:
Does the istio mtls influence the protocol selection?

Reply:
> It seems the current behavior is not work for pure http1.1 server either

Why do you think so? It works. We are using this for http1.1 server and also http2.0 server

Reply:
nm...I see what you are saying. If server only supports Http1.1, client can not use both http2/http1.1 - but that is not the usecase for `useClientProtocol` though.

Reply:
Yes currently if mtls is used (which it is by default) the server side will process inbound as raw tcp regardless of htto1 or http2. So there are actually two distinct issues at play here

Reply:
How about a client without sidecar communicate with a server with sidecar then? 
In this case. does it work?
client -> http2-> sidecar(server)->server(serve only http1.1)

Reply:
So this PR doesn't change that case at all. But what happens today depends on the protocol declared:
- http - works
- http2 - broken
- undeclared - broken since we use original protocol

Reply:
Yes, clear now.